xchange-therock/src/test/java/org/knowm/xchange/therock/dto/TheRockDtoTest.java

```diff
@@ -1,115 +0,0 @@
-package org.knowm.xchange.therock.dto;
-
-import static org.assertj.core.api.Assertions.assertThat;
-
-import com.fasterxml.jackson.databind.ObjectMapper;
-import java.io.IOException;
-import java.io.InputStream;
-import java.math.BigDecimal;
-import java.nio.file.Paths;
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
-import java.util.Arrays;
-import java.util.Date;
-import java.util.LinkedList;
-import org.junit.Test;
-import org.knowm.xchange.currency.CurrencyPair;
-import org.knowm.xchange.therock.TheRock;
-import org.knowm.xchange.therock.dto.account.TheRockBalance;
-import org.knowm.xchange.therock.dto.account.TheRockBalances;
-import org.knowm.xchange.therock.dto.account.TheRockWithdrawalResponse;
-import org.knowm.xchange.therock.dto.marketdata.TheRockTicker;
-import org.knowm.xchange.therock.dto.trade.TheRockOrder;
-import org.knowm.xchange.therock.dto.trade.TheRockOrders;
-
-public class TheRockDtoTest {
-
-  private static ObjectMapper mapper = new ObjectMapper();
-
-  private static <T> T parse(Class<T> theClass) throws IOException {
-    LinkedList<String> linkedList =
-        new LinkedList<>(
-            Arrays.asList(new Object() {}.getClass().getPackage().getName().split("\\.")));
-    linkedList.add(theClass.getSimpleName() + ".json");
-    String filename =
-        Paths.get(linkedList.removeFirst(), linkedList.toArray(new String[0])).toString();
-    return parse(filename, theClass);
-  }
-
-  private static <E> E parse(String filename, Class<E> type) throws java.io.IOException {
-    InputStream is = TheRockDtoTest.class.getResourceAsStream("/" + filename);
-    return mapper.readValue(is, type);
-  }
-
-  @Test
-  public void testTicker() throws Exception {
```

Comment:
so now we have an over complicated deserialization mechanism to the rock DTOs with less test coverage

restore this test, remove the deserializers, use Jackson annotation on DTOs

besides this looks unrelated to the futures feature - if so then separate this to a different PR

Reply:
removed, custom deserializer added to instrument.